### ISSUE TITLE
[SliverAppBar][Semantics] Voice over gets stuck when sliver app bar is collapsed.  

### ISSUE THREAD

**Issue by johnsonmh** [flutter/flutter#60331]
Internal: b/158621034

When you have an app with a `SliverAppBar` and a `SliverList` body, and you're using voiceover on iOS, scrolling to the bottom of the list and then back up will get the voice over stuck in the status bar. 

Additionally, an extra "Heading" node seems to be added. 

## Steps to Reproduce

Using the following example app:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(Demo());
}

class Demo extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: CustomScrollView(
          slivers: [
            SliverAppBar(
              pinned: true,
              expandedHeight: 200,
              flexibleSpace: Container(height: 100),
            ),
            SliverList(
              delegate: SliverChildListDelegate([
                for (var i = 0; i < 50; i ++)
                  Text('This is text. ' * 45),
              ]),
            ),
          ],
        ),
      ),
    );
  }
}
```

1. Open the above app with voiceover enabled on an iOS device.
1. Swipe to the right to navigate to the very bottom of the list. 
1. Swipe back up to the header, while it is still collapsed. 
1. Notice that the voice over tool gets stuck in the status bar, even when you swipe left and right, you can not get back to the page. 

**Expected results:** 
Swiping left to navigate back up the list will put you back in AppBar.

**Actual results:** 
Voice Over gets stuck in the status bar, back button, nor text in list, can not be navigated to.

Video of bug (error state begins at the ~8 second mark):
https://photos.app.goo.gl/bYAsCBz4ixdZmCcg9

<details>
  <summary>Flutter Doctor</summary>

```
```

```
```

```
[✓] Flutter (Channel master, 1.20.0-1.0.pre.135, on Mac OS X 10.15.5 19F96, locale en-US)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[!] Xcode - develop for iOS and macOS (Xcode 11.3)
    ✗ CocoaPods installed but not working.
        You appear to have CocoaPods installed but it is not working.
        This can happen if the version of Ruby that CocoaPods was installed with is different from the one being used to invoke it.
        This can usually be fixed by re-installing CocoaPods. For more info, see https://github.com/flutter/flutter/issues/14293.
      To re-install CocoaPods, run:
        sudo gem install cocoapods
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.0)
[✓] Android Studio (version 3.5)
[✓] Connected device (4 available)
```

</details>

**Comment by chunhtai** [comment]
It looks like we do send correct semantics tree over to the ios accessbility bridge, it is not clear to me why it focus a wrong rectangle when it focus the header.

The other problem is we do not scroll up enough when we swipe back up to reveal the full header.

**Comment by chunhtai** [comment]
It seems like the bug only happen when we change the accessbility tree dramatically due to the way we construct the semantics tree in scrollable.

The scrollable divide its children into two group ([the code ](https://github.com/flutter/flutter/blob/70a3dc0c61441a0a1652d69a9ec0a5ca558ae35d/packages/flutter/lib/src/widgets/scrollable.dart#L793))

The first group is children that are excluded from scrolling and the second group is children that are included. The sliver app bar is jumping between this two group. it is included when it is not collapsed, and excluded when it is collapsed.

![Untitled drawing (1)](https://user-images.githubusercontent.com/47866232/86819075-09033f80-c03c-11ea-9ba1-79be043f7f53.png)

it seems like when the native ios cannot handle such change when the focus is on the content in the customscrollview. I think this may be a native ios bug.

**Comment by liuyingshan** [comment]
> It looks like we do send correct semantics tree over to the ios accessbility bridge, it is not clear to me why it focus a wrong rectangle when it focus the header.
> 
> The other problem is we do not scroll up enough when we swipe back up to reveal the full header.

The other problem that we do not scroll up enough when we swipe back up to reveal the full header is still reproducible. Is there another issue tracking it?

**Comment by chunhtai** [comment]
@liuyingshan  I opened a issue to track the issue https://github.com/flutter/flutter/issues/61360
